sentry-android-core/src/main/java/io/sentry/android/core/EmptySecureContentProvider.java

```diff
@@ -0,0 +1,56 @@
+package io.sentry.android.core;
+
+import android.content.ContentProvider;
+import android.content.ContentValues;
+import android.database.Cursor;
+import android.net.Uri;
+import io.sentry.android.core.internal.util.ContentProviderSecurityChecker;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * A ContentProvider that does NOT store or provide any data for read or write operations.
+ *
+ * <p>This does not allow for overriding the abstract query, insert, update, and delete operations
+ * of the {@link ContentProvider}. Additionally, those functions are secure.
+ */
+@ApiStatus.Internal
+abstract class EmptySecureContentProvider extends ContentProvider {
+
+  private final ContentProviderSecurityChecker securityChecker =
+      new ContentProviderSecurityChecker();
```

Comment:
You will notice that I am not initializing this with a reference to `this` ContentProvider. If I did, the code would be a tiny bit shorter at function call-sites;

```java
securityChecker.checkPrivilegeEscalation();
```

It looks nice and is okay because there is no harm in saving a hard reference to the ContentProvider as long as we don't leak it. However, this would come at the price of having to assign the value of this in `onCreate`, which would require it being annotated with `@CallSuper`, and this not being `final`...

```java
EmptySecureContentProvider {

  private ContentProviderSecurityChecker securityChecker;

  @CallSuper
  @Override
  public boolean onCreate() {
    securityChecker = new ContentProviderSecurityChecker(this);
    return false;
  }
}
```

So.. I think that we don't need to go through all that trouble just so that we don't have to pas in a reference to the ContentProvider when calling `securityChecker.checkPrivilegeEscalation`.